worker/upgradesteps/worker.go

```diff
@@ -209,6 +209,9 @@ func (w *upgradesteps) run() error {
 			return errors.Trace(err)
 		}
 		w.isCaas = model.Type() == state.ModelTypeCAAS
+	} else {
+		// application tag for CAAS operator.
+		w.isCaas = names.IsValidApplication(w.agent.CurrentConfig().Tag().String())
```

Comment:
If we are doing htis can we just have one check for isCass that can be used for controller or not.
Having 2 checks is a bit clunky.

Reply:
yup, moved this to manifold then pass in